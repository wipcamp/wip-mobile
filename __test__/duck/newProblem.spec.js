/**
 * @jest-environment node
 */

import reducer, {
    setNewProblemTopic, 
    setNewProblemType, 
    setNewProblemDescription, 
    setNewProblemReportId
} from '../../ducks/NewProblem'

describe('new problem duck', () => {
    describe('action', () => {
        it('should create an action to set topic', () => {
            const topic = "Test Topic"
            const expectedAction = {
                type: 'SET_NEW_TOPIC',
                topic
            }
            expect(setNewProblemTopic(topic)).toEqual(expectedAction)
        })
    })

    describe('reducer', () => {

    })
})