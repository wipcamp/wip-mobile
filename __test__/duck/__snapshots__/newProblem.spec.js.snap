// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action setNewProblemDescription 1`] = `
Object {
  "description": "Test",
  "type": "SET_NEW_DESCRIPTION",
}
`;

exports[`action setNewProblemPriorityId 1`] = `
Object {
  "id": 1,
  "type": "SET_NEW_PRIORITY",
}
`;

exports[`action setNewProblemReportId 1`] = `
Object {
  "id": 1,
  "type": "SET_NEW_REPORTID",
}
`;

exports[`action setNewProblemTopic 1`] = `
Object {
  "topic": "Test",
  "type": "SET_NEW_TOPIC",
}
`;

exports[`action setNewProblemType 1`] = `
Object {
  "category": 1,
  "type": "SET_NEW_TYPE",
}
`;

exports[`reducer SET_NEW_DESCRIPTION 1`] = `
Object {
  "newProblem": Object {
    "description": "Test",
    "priority_id": 0,
    "problem_type_id": 0,
    "report_id": null,
    "topic": "",
  },
}
`;

exports[`reducer SET_NEW_PRIORITY 1`] = `
Object {
  "newProblem": Object {
    "description": "",
    "priority_id": 1,
    "problem_type_id": 0,
    "report_id": null,
    "topic": "",
  },
}
`;

exports[`reducer SET_NEW_REPORTID 1`] = `
Object {
  "newProblem": Object {
    "description": "",
    "priority_id": 0,
    "problem_type_id": 0,
    "report_id": 1,
    "topic": "",
  },
}
`;

exports[`reducer SET_NEW_TOPIC 1`] = `
Object {
  "newProblem": Object {
    "description": "",
    "priority_id": 0,
    "problem_type_id": 0,
    "report_id": null,
    "topic": "Test",
  },
}
`;

exports[`reducer SET_NEW_TYPE 1`] = `
Object {
  "newProblem": Object {
    "description": "",
    "priority_id": 0,
    "problem_type_id": 1,
    "report_id": null,
    "topic": "",
  },
}
`;

exports[`reducer default 1`] = `
Object {
  "categoryProblem": Array [],
  "newProblem": Object {
    "description": "",
    "priority_id": 0,
    "problem_type_id": 0,
    "report_id": null,
    "topic": "",
  },
  "notification": Array [],
  "priority": Array [],
  "problem": Array [],
  "timetable": Array [],
}
`;
